first commit

qiaoyuKs opened the pull request:
T2 competition, a joint team from Kuaishou Technology and Tsinghua University, modify the search process, indexing use the baseline index of the benchmark

harsha-simhadri commented:
> > @qiaoyuKs There seems to be some error in algos.yaml for ssnpp-1b dataset.
> > $ python3 run.py --algorithm kota-t2 --dataset ssnpp-1B downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries.u8bin -> data/FB_ssnpp/FB_ssnpp_public_queries.u8bin... [0.68 s] downloaded 24.41 MiB / 24.41 MiB at 35.95 MiB/s download finished in 0.68 s, total size 25600008 bytes downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries_1B_GT.rangeres -> data/FB_ssnpp/FB_ssnpp_public_queries_1B_GT.rangeres... [1.04 s] downloaded 59.18 MiB / 59.18 MiB at 57.15 MiB/s download finished in 1.04 s, total size 62054024 bytes Traceback (most recent call last): File "run.py", line 6, in main() File "/home/harshasi/big-ann-benchmarks/benchmark/main.py", line 173, in main args.definitions, dimension, args.dataset, distance, args.count) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 151, in get_definitions query_args = _generate_combinations(run_group["query-args"]) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 58, in _generate_combinations l = json.loads(args.strip()) File "/usr/lib/python3.6/json/**init**.py", line 354, in loads return _default_decoder.decode(s) File "/usr/lib/python3.6/json/decoder.py", line 339, in decode obj, end = self.raw_decode(s, idx=_w(s, 0).end()) File "/usr/lib/python3.6/json/decoder.py", line 355, in raw_decode obj, end = self.scan_once(s, idx) json.decoder.JSONDecodeError: Expecting ',' delimiter: line 4 column 43 (char 172)
> 
> @harsha-simhadri Sorry, we fixed the problem in ssnpp-1B query set ,it's a problem of comma, the ssnpp-1B will result in a better result, about 0.22, 0.06 increase than the benchmark

@qiaoyuKs The run for ssnpp-1B still fails. Could you please look into this. The error I see is

2021-11-07 08:57:53,426 - annb.f1c196f80c - INFO - Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ...
2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Metadata: #pts = 2712002176, #dims = 32643...
2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Error. File size mismatch. Actual size is 18446744073709551615 while expected size is  354111548124680 npts = 2712002176 dim = 32643 size of <T>= 4
2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - Traceback (most recent call last):
2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO -   File "run_algorithm.py", line 3, in <module>
2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO -     run_from_cmdline()
2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO -   File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline
2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO -     args.upload_index, args.download_index, args.blob_prefix, args.sas_string)
2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO -   File "/home/app/benchmark/runner.py", line 91, in run
2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO -     elif rebuild or not algo.load_index(dataset):
2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO -   File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index
2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO -     if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0):
2021-11-07 08:57:53,450 - annb.f1c196f80c - INFO - RuntimeError: Caught an unknown exception!
2021-11-07 08:57:56,061 - annb.f1c196f80c - ERROR - ['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}]
Trying to instantiate benchmark.algorithms.kota-t2.KoTa(['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}])
0
Running kota-t2 on ssnpp-1B
Got 100000 queries
dist_cmp function for _u8 uses slow implementation. Please contact gopalsr@microsoft.com if you need an AVX/AVX2 implementation.
Using AVX2 dist_cmp_float function.
Loading index
Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_compressed.bin ...
Metadata: #pts = 1000000000, #dims = 64...
tcmalloc: large alloc 64000000000 bytes == 0x34fc000 @  0x7f83a1a71887 0x7f83a2aca45f 0x7f83a2ad9c93 0x7f83a29c1e77 0x7f83a29f552f 0x7f83a29ce27e 0x566bec 0x50a533 0x50bf44 0x5096c8 0x50a3fd 0x50bf44 0x507cd4 0x509a00 0x50a3fd 0x50bf44 0x506044 0x50a3fd 0x50bf44 0x507cd4 0x50ae13 0x635262 0x635317 0x638acf 0x639671 0x4b0e40 0x7f83be5bfbf7 0x5b2f0a
Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ...
Metadata: #pts = 2712002176, #dims = 32643...
Error. File size mismatch. Actual size is 18446744073709551615 while expected size is  354111548124680 npts = 2712002176 dim = 32643 size of <T>= 4
Traceback (most recent call last):
  File "run_algorithm.py", line 3, in <module>
    run_from_cmdline()
  File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline
    args.upload_index, args.download_index, args.blob_prefix, args.sas_string)
  File "/home/app/benchmark/runner.py", line 91, in run
    elif rebuild or not algo.load_index(dataset):
  File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index
    if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0):
RuntimeError: Caught an unknown exception!

qiaoyuKs commented:
> > > @qiaoyuKs There seems to be some error in algos.yaml for ssnpp-1b dataset.
> > > $ python3 run.py --algorithm kota-t2 --dataset ssnpp-1B downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries.u8bin -> data/FB_ssnpp/FB_ssnpp_public_queries.u8bin... [0.68 s] downloaded 24.41 MiB / 24.41 MiB at 35.95 MiB/s download finished in 0.68 s, total size 25600008 bytes downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries_1B_GT.rangeres -> data/FB_ssnpp/FB_ssnpp_public_queries_1B_GT.rangeres... [1.04 s] downloaded 59.18 MiB / 59.18 MiB at 57.15 MiB/s download finished in 1.04 s, total size 62054024 bytes Traceback (most recent call last): File "run.py", line 6, in main() File "/home/harshasi/big-ann-benchmarks/benchmark/main.py", line 173, in main args.definitions, dimension, args.dataset, distance, args.count) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 151, in get_definitions query_args = _generate_combinations(run_group["query-args"]) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 58, in _generate_combinations l = json.loads(args.strip()) File "/usr/lib/python3.6/json/**init**.py", line 354, in loads return _default_decoder.decode(s) File "/usr/lib/python3.6/json/decoder.py", line 339, in decode obj, end = self.raw_decode(s, idx=_w(s, 0).end()) File "/usr/lib/python3.6/json/decoder.py", line 355, in raw_decode obj, end = self.scan_once(s, idx) json.decoder.JSONDecodeError: Expecting ',' delimiter: line 4 column 43 (char 172)
> > 
> > 
> > @harsha-simhadri Sorry, we fixed the problem in ssnpp-1B query set ,it's a problem of comma, the ssnpp-1B will result in a better result, about 0.22, 0.06 increase than the benchmark
> 
> @qiaoyuKs The run for ssnpp-1B still fails. Could you please look into this. The error I see is
> 
> 2021-11-07 08:57:53,426 - annb.f1c196f80c - INFO - Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ... 2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Metadata: #pts = 2712002176, #dims = 32643... 2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Error. File size mismatch. Actual size is 18446744073709551615 while expected size is 354111548124680 npts = 2712002176 dim = 32643 size of = 4 2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - Traceback (most recent call last): 2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - File "run_algorithm.py", line 3, in 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - run_from_cmdline() 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - args.upload_index, args.download_index, args.blob_prefix, args.sas_string) 2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO - File "/home/app/benchmark/runner.py", line 91, in run 2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO - elif rebuild or not algo.load_index(dataset): 2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO - File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index 2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO - if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0): 2021-11-07 08:57:53,450 - annb.f1c196f80c - INFO - RuntimeError: Caught an unknown exception! 2021-11-07 08:57:56,061 - annb.f1c196f80c - ERROR - ['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}] Trying to instantiate benchmark.algorithms.kota-t2.KoTa(['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}]) 0 Running kota-t2 on ssnpp-1B Got 100000 queries dist_cmp function for _u8 uses slow implementation. Please contact [gopalsr@microsoft.com](mailto:gopalsr@microsoft.com) if you need an AVX/AVX2 implementation. Using AVX2 dist_cmp_float function. Loading index Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_compressed.bin ... Metadata: #pts = 1000000000, #dims = 64... tcmalloc: large alloc 64000000000 bytes == 0x34fc000 @ 0x7f83a1a71887 0x7f83a2aca45f 0x7f83a2ad9c93 0x7f83a29c1e77 0x7f83a29f552f 0x7f83a29ce27e 0x566bec 0x50a533 0x50bf44 0x5096c8 0x50a3fd 0x50bf44 0x507cd4 0x509a00 0x50a3fd 0x50bf44 0x506044 0x50a3fd 0x50bf44 0x507cd4 0x50ae13 0x635262 0x635317 0x638acf 0x639671 0x4b0e40 0x7f83be5bfbf7 0x5b2f0a Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ... Metadata: #pts = 2712002176, #dims = 32643... Error. File size mismatch. Actual size is 18446744073709551615 while expected size is 354111548124680 npts = 2712002176 dim = 32643 size of = 4 Traceback (most recent call last): File "run_algorithm.py", line 3, in run_from_cmdline() File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline args.upload_index, args.download_index, args.blob_prefix, args.sas_string) File "/home/app/benchmark/runner.py", line 91, in run elif rebuild or not algo.load_index(dataset): File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0): RuntimeError: Caught an unknown exception!

@harsha-simhadri We checked the code and it could run in our azure machine, could you provide more information about the indices file size, after we download the indices from the url "https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100", we got the following

-rw-r--r-- 1 root root 682666672128 Nov  8 02:44 R100_L100_B60_M100_disk.index
-rw-r--r-- 1 root root  64000000008 Nov  8 02:17 R100_L100_B60_M100_pq_compressed.bin
-rw-r--r-- 1 root root       599264 Nov  8 02:15 R100_L100_B60_M100_sample_ids.bin
-rw-r--r-- 1 root root     38352392 Nov  8 02:15 R100_L100_B60_M100_sample_data.bin
-rw-r--r-- 1 root root         1032 Nov  8 02:15 R100_L100_B60_M100_pq_pivots.bin_rearrangement_perm.bin
-rw-r--r-- 1 root root          268 Nov  8 02:15 R100_L100_B60_M100_pq_pivots.bin_chunk_offsets.bin
-rw-r--r-- 1 root root         1032 Nov  8 02:15 R100_L100_B60_M100_pq_pivots.bin_centroid.bin
-rw-r--r-- 1 root root       262152 Nov  8 02:15 R100_L100_B60_M100_pq_pivots.bin

did you get the same file?

qiaoyuKs commented:
> > > @qiaoyuKs There seems to be some error in algos.yaml for ssnpp-1b dataset.
> > > $ python3 run.py --algorithm kota-t2 --dataset ssnpp-1B downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries.u8bin -> data/FB_ssnpp/FB_ssnpp_public_queries.u8bin... [0.68 s] downloaded 24.41 MiB / 24.41 MiB at 35.95 MiB/s download finished in 0.68 s, total size 25600008 bytes downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries_1B_GT.rangeres -> data/FB_ssnpp/FB_ssnpp_public_queries_1B_GT.rangeres... [1.04 s] downloaded 59.18 MiB / 59.18 MiB at 57.15 MiB/s download finished in 1.04 s, total size 62054024 bytes Traceback (most recent call last): File "run.py", line 6, in main() File "/home/harshasi/big-ann-benchmarks/benchmark/main.py", line 173, in main args.definitions, dimension, args.dataset, distance, args.count) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 151, in get_definitions query_args = _generate_combinations(run_group["query-args"]) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 58, in _generate_combinations l = json.loads(args.strip()) File "/usr/lib/python3.6/json/**init**.py", line 354, in loads return _default_decoder.decode(s) File "/usr/lib/python3.6/json/decoder.py", line 339, in decode obj, end = self.raw_decode(s, idx=_w(s, 0).end()) File "/usr/lib/python3.6/json/decoder.py", line 355, in raw_decode obj, end = self.scan_once(s, idx) json.decoder.JSONDecodeError: Expecting ',' delimiter: line 4 column 43 (char 172)
> > 
> > 
> > @harsha-simhadri Sorry, we fixed the problem in ssnpp-1B query set ,it's a problem of comma, the ssnpp-1B will result in a better result, about 0.22, 0.06 increase than the benchmark
> 
> @qiaoyuKs The run for ssnpp-1B still fails. Could you please look into this. The error I see is
> 
> 2021-11-07 08:57:53,426 - annb.f1c196f80c - INFO - Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ... 2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Metadata: #pts = 2712002176, #dims = 32643... 2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Error. File size mismatch. Actual size is 18446744073709551615 while expected size is 354111548124680 npts = 2712002176 dim = 32643 size of = 4 2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - Traceback (most recent call last): 2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - File "run_algorithm.py", line 3, in 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - run_from_cmdline() 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - args.upload_index, args.download_index, args.blob_prefix, args.sas_string) 2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO - File "/home/app/benchmark/runner.py", line 91, in run 2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO - elif rebuild or not algo.load_index(dataset): 2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO - File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index 2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO - if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0): 2021-11-07 08:57:53,450 - annb.f1c196f80c - INFO - RuntimeError: Caught an unknown exception! 2021-11-07 08:57:56,061 - annb.f1c196f80c - ERROR - ['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}] Trying to instantiate benchmark.algorithms.kota-t2.KoTa(['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}]) 0 Running kota-t2 on ssnpp-1B Got 100000 queries dist_cmp function for _u8 uses slow implementation. Please contact [gopalsr@microsoft.com](mailto:gopalsr@microsoft.com) if you need an AVX/AVX2 implementation. Using AVX2 dist_cmp_float function. Loading index Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_compressed.bin ... Metadata: #pts = 1000000000, #dims = 64... tcmalloc: large alloc 64000000000 bytes == 0x34fc000 @ 0x7f83a1a71887 0x7f83a2aca45f 0x7f83a2ad9c93 0x7f83a29c1e77 0x7f83a29f552f 0x7f83a29ce27e 0x566bec 0x50a533 0x50bf44 0x5096c8 0x50a3fd 0x50bf44 0x507cd4 0x509a00 0x50a3fd 0x50bf44 0x506044 0x50a3fd 0x50bf44 0x507cd4 0x50ae13 0x635262 0x635317 0x638acf 0x639671 0x4b0e40 0x7f83be5bfbf7 0x5b2f0a Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ... Metadata: #pts = 2712002176, #dims = 32643... Error. File size mismatch. Actual size is 18446744073709551615 while expected size is 354111548124680 npts = 2712002176 dim = 32643 size of = 4 Traceback (most recent call last): File "run_algorithm.py", line 3, in run_from_cmdline() File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline args.upload_index, args.download_index, args.blob_prefix, args.sas_string) File "/home/app/benchmark/runner.py", line 91, in run elif rebuild or not algo.load_index(dataset): File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0): RuntimeError: Caught an unknown exception!

@harsha-simhadri could you provide the file size of R100_L100_B60_M100_pq_pivots.bin you used?

harsha-simhadri commented:
> > > > @qiaoyuKs There seems to be some error in algos.yaml for ssnpp-1b dataset.
> > > > $ python3 run.py --algorithm kota-t2 --dataset ssnpp-1B downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries.u8bin -> data/FB_ssnpp/FB_ssnpp_public_queries.u8bin... [0.68 s] downloaded 24.41 MiB / 24.41 MiB at 35.95 MiB/s download finished in 0.68 s, total size 25600008 bytes downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries_1B_GT.rangeres -> data/FB_ssnpp/FB_ssnpp_public_queries_1B_GT.rangeres... [1.04 s] downloaded 59.18 MiB / 59.18 MiB at 57.15 MiB/s download finished in 1.04 s, total size 62054024 bytes Traceback (most recent call last): File "run.py", line 6, in main() File "/home/harshasi/big-ann-benchmarks/benchmark/main.py", line 173, in main args.definitions, dimension, args.dataset, distance, args.count) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 151, in get_definitions query_args = _generate_combinations(run_group["query-args"]) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 58, in _generate_combinations l = json.loads(args.strip()) File "/usr/lib/python3.6/json/**init**.py", line 354, in loads return _default_decoder.decode(s) File "/usr/lib/python3.6/json/decoder.py", line 339, in decode obj, end = self.raw_decode(s, idx=_w(s, 0).end()) File "/usr/lib/python3.6/json/decoder.py", line 355, in raw_decode obj, end = self.scan_once(s, idx) json.decoder.JSONDecodeError: Expecting ',' delimiter: line 4 column 43 (char 172)
> > > 
> > > 
> > > @harsha-simhadri Sorry, we fixed the problem in ssnpp-1B query set ,it's a problem of comma, the ssnpp-1B will result in a better result, about 0.22, 0.06 increase than the benchmark
> > 
> > 
> > @qiaoyuKs The run for ssnpp-1B still fails. Could you please look into this. The error I see is
> > 2021-11-07 08:57:53,426 - annb.f1c196f80c - INFO - Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ... 2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Metadata: #pts = 2712002176, #dims = 32643... 2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Error. File size mismatch. Actual size is 18446744073709551615 while expected size is 354111548124680 npts = 2712002176 dim = 32643 size of = 4 2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - Traceback (most recent call last): 2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - File "run_algorithm.py", line 3, in 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - run_from_cmdline() 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - args.upload_index, args.download_index, args.blob_prefix, args.sas_string) 2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO - File "/home/app/benchmark/runner.py", line 91, in run 2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO - elif rebuild or not algo.load_index(dataset): 2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO - File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index 2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO - if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0): 2021-11-07 08:57:53,450 - annb.f1c196f80c - INFO - RuntimeError: Caught an unknown exception! 2021-11-07 08:57:56,061 - annb.f1c196f80c - ERROR - ['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}] Trying to instantiate benchmark.algorithms.kota-t2.KoTa(['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}]) 0 Running kota-t2 on ssnpp-1B Got 100000 queries dist_cmp function for _u8 uses slow implementation. Please contact [gopalsr@microsoft.com](mailto:gopalsr@microsoft.com) if you need an AVX/AVX2 implementation. Using AVX2 dist_cmp_float function. Loading index Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_compressed.bin ... Metadata: #pts = 1000000000, #dims = 64... tcmalloc: large alloc 64000000000 bytes == 0x34fc000 @ 0x7f83a1a71887 0x7f83a2aca45f 0x7f83a2ad9c93 0x7f83a29c1e77 0x7f83a29f552f 0x7f83a29ce27e 0x566bec 0x50a533 0x50bf44 0x5096c8 0x50a3fd 0x50bf44 0x507cd4 0x509a00 0x50a3fd 0x50bf44 0x506044 0x50a3fd 0x50bf44 0x507cd4 0x50ae13 0x635262 0x635317 0x638acf 0x639671 0x4b0e40 0x7f83be5bfbf7 0x5b2f0a Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ... Metadata: #pts = 2712002176, #dims = 32643... Error. File size mismatch. Actual size is 18446744073709551615 while expected size is 354111548124680 npts = 2712002176 dim = 32643 size of = 4 Traceback (most recent call last): File "run_algorithm.py", line 3, in run_from_cmdline() File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline args.upload_index, args.download_index, args.blob_prefix, args.sas_string) File "/home/app/benchmark/runner.py", line 91, in run elif rebuild or not algo.load_index(dataset): File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0): RuntimeError: Caught an unknown exception!
> 
> @harsha-simhadri could you provide the file size of R100_L100_B60_M100_pq_pivots.bin you used?

@qiaoyuKs size below
 -rw-r--r-- 1 root root 262152 Nov 13 14:47 data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin

Also, I am able to load and run the first 4 query args for ssnpp-1B, but the docker raises exception 137 after that. Your memory requirement might be running too close to the limit. You might want to recalibrate your query args to leave more memory for OS and other background processes

Load index success.
Loaded existing index
Index memory footprint:  63001892.0
Starting query
Running query argument group 1 of 10...
Run 1/5...
Run 2/5...
Run 3/5...
Run 4/5...
Run 5/5...
Running query argument group 2 of 10...
Run 1/5...
Run 2/5...
Run 3/5...
Run 4/5...
Run 5/5...
Running query argument group 3 of 10...
Run 1/5...
Run 2/5...
Run 3/5...
Run 4/5...
Run 5/5...
Running query argument group 4 of 10...
Run 1/5...
Run 2/5...
Run 3/5...
Run 4/5...
Run 5/5...
Running query argument group 5 of 10...
Run 1/5...

2021-11-13 15:41:35,188 - annb.9ecde6d6d9 - ERROR - Child process for container 9ecde6d6d9 raised exception 137

For the 4 query args that completed, I see the following results.

kota-t2,DiskANN,ssnpp-1B,60000,2237.9455403144407,1572.170728,1000000.0,63001892.0,28151.664490972365,31.02686,2363.41259,0.1557059316298228
kota-t2,DiskANN,ssnpp-1B,60000,2329.023940719631,1591.910644,1000000.0,63001892.0,27050.770453022236,30.04728,2410.80347,0.14927481771828133
kota-t2,DiskANN,ssnpp-1B,60000,2746.5274514348266,1328.769594,1000000.0,63001892.0,22938.74469271621,24.35615,1937.67508,0.1332291476357355
kota-t2,DiskANN,ssnpp-1B,60000,1968.80

qiaoyuKs commented:
> > > > > @qiaoyuKs There seems to be some error in algos.yaml for ssnpp-1b dataset.
> > > > > $ python3 run.py --algorithm kota-t2 --dataset ssnpp-1B downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries.u8bin -> data/FB_ssnpp/FB_ssnpp_public_queries.u8bin... [0.68 s] downloaded 24.41 MiB / 24.41 MiB at 35.95 MiB/s download finished in 0.68 s, total size 25600008 bytes downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries_1B_GT.rangeres -> data/FB_ssnpp/FB_ssnpp_public_queries_1B_GT.rangeres... [1.04 s] downloaded 59.18 MiB / 59.18 MiB at 57.15 MiB/s download finished in 1.04 s, total size 62054024 bytes Traceback (most recent call last): File "run.py", line 6, in main() File "/home/harshasi/big-ann-benchmarks/benchmark/main.py", line 173, in main args.definitions, dimension, args.dataset, distance, args.count) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 151, in get_definitions query_args = _generate_combinations(run_group["query-args"]) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 58, in _generate_combinations l = json.loads(args.strip()) File "/usr/lib/python3.6/json/**init**.py", line 354, in loads return _default_decoder.decode(s) File "/usr/lib/python3.6/json/decoder.py", line 339, in decode obj, end = self.raw_decode(s, idx=_w(s, 0).end()) File "/usr/lib/python3.6/json/decoder.py", line 355, in raw_decode obj, end = self.scan_once(s, idx) json.decoder.JSONDecodeError: Expecting ',' delimiter: line 4 column 43 (char 172)
> > > > 
> > > > 
> > > > @harsha-simhadri Sorry, we fixed the problem in ssnpp-1B query set ,it's a problem of comma, the ssnpp-1B will result in a better result, about 0.22, 0.06 increase than the benchmark
> > > 
> > > 
> > > @qiaoyuKs The run for ssnpp-1B still fails. Could you please look into this. The error I see is
> > > 2021-11-07 08:57:53,426 - annb.f1c196f80c - INFO - Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ... 2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Metadata: #pts = 2712002176, #dims = 32643... 2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Error. File size mismatch. Actual size is 18446744073709551615 while expected size is 354111548124680 npts = 2712002176 dim = 32643 size of = 4 2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - Traceback (most recent call last): 2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - File "run_algorithm.py", line 3, in 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - run_from_cmdline() 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - args.upload_index, args.download_index, args.blob_prefix, args.sas_string) 2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO - File "/home/app/benchmark/runner.py", line 91, in run 2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO - elif rebuild or not algo.load_index(dataset): 2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO - File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index 2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO - if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0): 2021-11-07 08:57:53,450 - annb.f1c196f80c - INFO - RuntimeError: Caught an unknown exception! 2021-11-07 08:57:56,061 - annb.f1c196f80c - ERROR - ['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}] Trying to instantiate benchmark.algorithms.kota-t2.KoTa(['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}]) 0 Running kota-t2 on ssnpp-1B Got 100000 queries dist_cmp function for _u8 uses slow implementation. Please contact [gopalsr@microsoft.com](mailto:gopalsr@microsoft.com) if you need an AVX/AVX2 implementation. Using AVX2 dist_cmp_float function. Loading index Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_compressed.bin ... Metadata: #pts = 1000000000, #dims = 64... tcmalloc: large alloc 64000000000 bytes == 0x34fc000 @ 0x7f83a1a71887 0x7f83a2aca45f 0x7f83a2ad9c93 0x7f83a29c1e77 0x7f83a29f552f 0x7f83a29ce27e 0x566bec 0x50a533 0x50bf44 0x5096c8 0x50a3fd 0x50bf44 0x507cd4 0x509a00 0x50a3fd 0x50bf44 0x506044 0x50a3fd 0x50bf44 0x507cd4 0x50ae13 0x635262 0x635317 0x638acf 0x639671 0x4b0e40 0x7f83be5bfbf7 0x5b2f0a Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ... Metadata: #pts = 2712002176, #dims = 32643... Error. File size mismatch. Actual size is 18446744073709551615 while expected size is 354111548124680 npts = 2712002176 dim = 32643 size of = 4 Traceback (most recent call last): File "run_algorithm.py", line 3, in run_from_cmdline() File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline args.upload_index, args.download_index, args.blob_prefix, args.sas_string) File "/home/app/benchmark/runner.py", line 91, in run elif rebuild or not algo.load_index(dataset): File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0): RuntimeError: Caught an unknown exception!
> > 
> > 
> > @harsha-simhadri could you provide the file size of R100_L100_B60_M100_pq_pivots.bin you used?
> 
> @qiaoyuKs size below -rw-r--r-- 1 root root 262152 Nov 13 14:47 data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin
> 
> Also, I am able to load and run the first 4 query args for ssnpp-1B, but the docker raises exception 137 after that. Your memory requirement might be running too close to the limit. You might want to recalibrate your query args to leave more memory for OS and other background processes
> 
> Load index success. Loaded existing index Index memory footprint: 63001892.0 Starting query Running query argument group 1 of 10... Run 1/5... Run 2/5... Run 3/5... Run 4/5... Run 5/5... Running query argument group 2 of 10... Run 1/5... Run 2/5... Run 3/5... Run 4/5... Run 5/5... Running query argument group 3 of 10... Run 1/5... Run 2/5... Run 3/5... Run 4/5... Run 5/5... Running query argument group 4 of 10... Run 1/5... Run 2/5... Run 3/5... Run 4/5... Run 5/5... Running query argument group 5 of 10... Run 1/5...
> 
> 2021-11-13 15:41:35,188 - annb.9ecde6d6d9 - ERROR - Child process for container 9ecde6d6d9 raised exception 137
> 
> For the 4 query args that completed, I see the following results.
> 
> kota-t2,DiskANN,ssnpp-1B,60000,2237.9455403144407,1572.170728,1000000.0,63001892.0,28151.664490972365,31.02686,2363.41259,0.1557059316298228 kota-t2,DiskANN,ssnpp-1B,60000,2329.023940719631,1591.910644,1000000.0,63001892.0,27050.770453022236,30.04728,2410.80347,0.14927481771828133 kota-t2,DiskANN,ssnpp-1B,60000,2746.5274514348266,1328.769594,1000000.0,63001892.0,22938.74469271621,24.35615,1937.67508,0.1332291476357355 kota-t2,DiskANN,ssnpp-1B,60000,1968.80

@harsha-simhadri Actually we couldn't reproduce the problem in our azure machine before, we have modified our source code of our algorithm to decrease the additional memory we introduced compared with diskann algorithm, you can try that. Another question is that we have used up  the Azure Sponsorship, is there any additional code or azure machine we can use?

qiaoyuKs commented:
> > > > > @qiaoyuKs There seems to be some error in algos.yaml for ssnpp-1b dataset.
> > > > > $ python3 run.py --algorithm kota-t2 --dataset ssnpp-1B downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries.u8bin -> data/FB_ssnpp/FB_ssnpp_public_queries.u8bin... [0.68 s] downloaded 24.41 MiB / 24.41 MiB at 35.95 MiB/s download finished in 0.68 s, total size 25600008 bytes downloading https://dl.fbaipublicfiles.com/billion-scale-ann-benchmarks/FB_ssnpp_public_queries_1B_GT.rangeres -> data/FB_ssnpp/FB_ssnpp_public_queries_1B_GT.rangeres... [1.04 s] downloaded 59.18 MiB / 59.18 MiB at 57.15 MiB/s download finished in 1.04 s, total size 62054024 bytes Traceback (most recent call last): File "run.py", line 6, in main() File "/home/harshasi/big-ann-benchmarks/benchmark/main.py", line 173, in main args.definitions, dimension, args.dataset, distance, args.count) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 151, in get_definitions query_args = _generate_combinations(run_group["query-args"]) File "/home/harshasi/big-ann-benchmarks/benchmark/algorithms/definitions.py", line 58, in _generate_combinations l = json.loads(args.strip()) File "/usr/lib/python3.6/json/**init**.py", line 354, in loads return _default_decoder.decode(s) File "/usr/lib/python3.6/json/decoder.py", line 339, in decode obj, end = self.raw_decode(s, idx=_w(s, 0).end()) File "/usr/lib/python3.6/json/decoder.py", line 355, in raw_decode obj, end = self.scan_once(s, idx) json.decoder.JSONDecodeError: Expecting ',' delimiter: line 4 column 43 (char 172)
> > > > 
> > > > 
> > > > @harsha-simhadri Sorry, we fixed the problem in ssnpp-1B query set ,it's a problem of comma, the ssnpp-1B will result in a better result, about 0.22, 0.06 increase than the benchmark
> > > 
> > > 
> > > @qiaoyuKs The run for ssnpp-1B still fails. Could you please look into this. The error I see is
> > > 2021-11-07 08:57:53,426 - annb.f1c196f80c - INFO - Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ... 2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Metadata: #pts = 2712002176, #dims = 32643... 2021-11-07 08:57:53,427 - annb.f1c196f80c - INFO - Error. File size mismatch. Actual size is 18446744073709551615 while expected size is 354111548124680 npts = 2712002176 dim = 32643 size of = 4 2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - Traceback (most recent call last): 2021-11-07 08:57:53,446 - annb.f1c196f80c - INFO - File "run_algorithm.py", line 3, in 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - run_from_cmdline() 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline 2021-11-07 08:57:53,447 - annb.f1c196f80c - INFO - args.upload_index, args.download_index, args.blob_prefix, args.sas_string) 2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO - File "/home/app/benchmark/runner.py", line 91, in run 2021-11-07 08:57:53,448 - annb.f1c196f80c - INFO - elif rebuild or not algo.load_index(dataset): 2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO - File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index 2021-11-07 08:57:53,449 - annb.f1c196f80c - INFO - if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0): 2021-11-07 08:57:53,450 - annb.f1c196f80c - INFO - RuntimeError: Caught an unknown exception! 2021-11-07 08:57:56,061 - annb.f1c196f80c - ERROR - ['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}] Trying to instantiate benchmark.algorithms.kota-t2.KoTa(['euclidean', {'R': 100, 'L': 100, 'B': 60, 'M': 100, 'C': 500000, 'CM': 2, 'url': 'https://comp21storage.blob.core.windows.net/publiccontainer/comp21/diskann-T2-baseline-indices/fbssnpp-1B/R100_L100_B60_M100'}]) 0 Running kota-t2 on ssnpp-1B Got 100000 queries dist_cmp function for _u8 uses slow implementation. Please contact [gopalsr@microsoft.com](mailto:gopalsr@microsoft.com) if you need an AVX/AVX2 implementation. Using AVX2 dist_cmp_float function. Loading index Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_compressed.bin ... Metadata: #pts = 1000000000, #dims = 64... tcmalloc: large alloc 64000000000 bytes == 0x34fc000 @ 0x7f83a1a71887 0x7f83a2aca45f 0x7f83a2ad9c93 0x7f83a29c1e77 0x7f83a29f552f 0x7f83a29ce27e 0x566bec 0x50a533 0x50bf44 0x5096c8 0x50a3fd 0x50bf44 0x507cd4 0x509a00 0x50a3fd 0x50bf44 0x506044 0x50a3fd 0x50bf44 0x507cd4 0x50ae13 0x635262 0x635317 0x638acf 0x639671 0x4b0e40 0x7f83be5bfbf7 0x5b2f0a Reading bin file /home/app/data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin ... Metadata: #pts = 2712002176, #dims = 32643... Error. File size mismatch. Actual size is 18446744073709551615 while expected size is 354111548124680 npts = 2712002176 dim = 32643 size of = 4 Traceback (most recent call last): File "run_algorithm.py", line 3, in run_from_cmdline() File "/home/app/benchmark/runner.py", line 228, in run_from_cmdline args.upload_index, args.download_index, args.blob_prefix, args.sas_string) File "/home/app/benchmark/runner.py", line 91, in run elif rebuild or not algo.load_index(dataset): File "/home/app/benchmark/algorithms/kota-t2.py", line 221, in load_index if (self.index.load_index(index_path, diskannpy.omp_get_max_threads(), num_nodes_to_cache, self.cache_mechanism, False, True, 4, "int") == 0): RuntimeError: Caught an unknown exception!
> > 
> > 
> > @harsha-simhadri could you provide the file size of R100_L100_B60_M100_pq_pivots.bin you used?
> 
> @qiaoyuKs size below -rw-r--r-- 1 root root 262152 Nov 13 14:47 data/indices/T2/DiskANN/SSNPPDataset-1000000000/R100_L100_B60_M100/R100_L100_B60_M100_pq_pivots.bin
> 
> Also, I am able to load and run the first 4 query args for ssnpp-1B, but the docker raises exception 137 after that. Your memory requirement might be running too close to the limit. You might want to recalibrate your query args to leave more memory for OS and other background processes
> 
> Load index success. Loaded existing index Index memory footprint: 63001892.0 Starting query Running query argument group 1 of 10... Run 1/5... Run 2/5... Run 3/5... Run 4/5... Run 5/5... Running query argument group 2 of 10... Run 1/5... Run 2/5... Run 3/5... Run 4/5... Run 5/5... Running query argument group 3 of 10... Run 1/5... Run 2/5... Run 3/5... Run 4/5... Run 5/5... Running query argument group 4 of 10... Run 1/5... Run 2/5... Run 3/5... Run 4/5... Run 5/5... Running query argument group 5 of 10... Run 1/5...
> 
> 2021-11-13 15:41:35,188 - annb.9ecde6d6d9 - ERROR - Child process for container 9ecde6d6d9 raised exception 137
> 
> For the 4 query args that completed, I see the following results.
> 
> kota-t2,DiskANN,ssnpp-1B,60000,2237.9455403144407,1572.170728,1000000.0,63001892.0,28151.664490972365,31.02686,2363.41259,0.1557059316298228 kota-t2,DiskANN,ssnpp-1B,60000,2329.023940719631,1591.910644,1000000.0,63001892.0,27050.770453022236,30.04728,2410.80347,0.14927481771828133 kota-t2,DiskANN,ssnpp-1B,60000,2746.5274514348266,1328.769594,1000000.0,63001892.0,22938.74469271621,24.35615,1937.67508,0.1332291476357355 kota-t2,DiskANN,ssnpp-1B,60000,1968.80
@harsha-simhadri  Have you tried the algorithm in test set?

harsha-simhadri commented:
@qiaoyuKs  I am able to run only the first 3 query arg combinations of ssnpp-1B before the docker runs out of memory (exception 137). You may want to be conservative with memory usage.

For the query combinations that ran, I see a operating point with AP of 0.182 that's above QPS threshold. We will publish this for the competition.

If you could please rebase your code and squash your changes into one commit, I will get started with merging this PR. 

You are welcome to keep improving your algorithm and updating your submission post NeurIPS.

qiaoyuKs commented:
> @qiaoyuKs I am able to run only the first 3 query arg combinations of ssnpp-1B before the docker runs out of memory (exception 137). You may want to be conservative with memory usage.
> 
> For the query combinations that ran, I see a operating point with AP of 0.182 that's above QPS threshold. We will publish this for the competition.
> 
> If you could please rebase your code and squash your changes into one commit, I will get started with merging this PR.
> 
> You are welcome to keep improving your algorithm and updating your submission post NeurIPS.

@harsha-simhadri There are so many conflicts to deal with, could I open a new PR and submit the code?

harsha-simhadri commented:
> > @qiaoyuKs I am able to run only the first 3 query arg combinations of ssnpp-1B before the docker runs out of memory (exception 137). You may want to be conservative with memory usage.
> > For the query combinations that ran, I see a operating point with AP of 0.182 that's above QPS threshold. We will publish this for the competition.
> > If you could please rebase your code and squash your changes into one commit, I will get started with merging this PR.
> > You are welcome to keep improving your algorithm and updating your submission post NeurIPS.
> 
> @harsha-simhadri There are so many conflicts to deal with, could I open a new PR and submit the code?

@qiaoyuKs. yes please. Could you please resolve conflicts and let's merge this to main
